src/plugins/intel_gpu/src/graph/graph_optimizer/add_required_reorders.cpp

```diff
@@ -283,6 +283,10 @@ void add_required_reorders::run(program& p) {
                     new_reorder_node.recalc_output_layout(true);
                 }
             }
+
+            if (!usr->is_all_valid_output_layouts()) {
+                usr->recalc_output_layout(false);
```

Comment:
I think it would be more straightforward if you just put this line into the `if(requires_alignment)` block.

Reply:
Moved the changes into the block you mentioned.